staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -99,6 +102,15 @@ func TestProgressNotify(t *testing.T) {
 	storagetesting.RunOptionalTestProgressNotify(ctx, t, store)
 }
 
+func TestSendInitialEventsBackwardCompatibility(t *testing.T) {
```

Comment:
The idea is that we're trying as many tests implementation-agnostic as possible and run the exact same tests against both etcd3 and watchcache.

It seems to be one of those tests, so let's implement it in pkg/storage/testing/
and run it for both etcd3 and watchcache (pkg/storage/tests/cacher_test.go).

Reply:
Yes - see above, all the tests above are following this pattern.

Reply:
okay, I think I have it. PTAL